File: 03-tidy/code.py
import pandas as pd

URL = "https://raw.githubusercontent.com/cmrivers/ebola/refs/heads/master/country_timeseries.csv"
ebola = pd.read_csv(URL)
ebola

ebola_long = ebola.melt(
    id_vars=["Date", "Day"],
    var_name="cd_country",
    value_name="count",
)
ebola_long


# don't need to know
ebola_long[["case_death", "country"]] = ebola_long.cd_country.str.split(
    "_",
    expand=True,
)
ebola_long


ebola_long.groupby(["case_death", "country"])["count"].max().reset_index()


ebola_tidy = (
    ebola_long.drop(columns="cd_country")  # dropping a column
    .pivot(index=["Date", "Day", "country"], columns="case_death", values="count")
    .reset_index()
    .dropna()
)
ebola_tidy

# final clean up of names
# don't need to know
ebola_tidy.columns.name = None
ebola_tidy

File: 03-tidy/code.R
library(tidyverse)

URL <- "https://raw.githubusercontent.com/cmrivers/ebola/refs/heads/master/country_timeseries.csv"
ebola <- read_csv(URL)

ebola_long <- ebola |>
  pivot_longer(
    cols = -c(Date, Day),
    names_to = "cd_country",
    values_to = "count"
  )

ebola_long <- ebola_long |>
  separate(
    cd_country,
    into = c("case_death", "country"),
    sep = "_"
  )

ebola_long |>
  summarize(max_count = max(count, na.rm = TRUE), .by = c(case_death, country))

ebola_long |>
  group_by(case_death, country) |>
  summarize(max = max(count, na.rm = TRUE))

ebola_tidy <- ebola_long |>
  pivot_wider(
    id_cols = c(Date, Day, country),
    names_from = case_death,
    values_from = count
  ) |>
  drop_na()

ebola_tidy

ebola_tidy |>
  group_by(country) |>
  summarize(
    max_case = max(Cases, na.rm = TRUE),
    max_death = max(Deaths, na.rm = TRUE)
  )


mtcars |>
  group_by(cyl) |>
    summarise(
      disp = mean(disp),
      hp = mean(hp),
      mpg = mean(mpg)
  )


# vs - engine shape v or straight
#vs: Engine shape (0 = V-shape, 1 = Straight)
#am: Transmission (0 = Automatic, 1 = Manual)

mtcars |>
  group_by(vs, am) |>
  summarise(n = n())

mtcars |>
  group_by(cyl) |>
    summarise(
      disp = mean(disp),
      hp = mean(hp),
      mpg = mean(mpg),
      n = n()
  )
